ddtrace/tracer/span.go

```diff
@@ -130,21 +132,32 @@ func (s *span) SetTag(key string, value interface{}) {
 // setTagError sets the error tag. It accounts for various valid scenarios.
 // This method is not safe for concurrent use.
 func (s *span) setTagError(value interface{}, cfg errorConfig) {
+	setError := func(yes bool) {
```

Comment:
Why do we set errors here, instead of when closing the span?

Reply:
This code path gets called when setting an error, regardless of where: by means of setting a tag, or, when closing the span.